[2024-10-29T15:16:46.958+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-10-29T15:16:46.975+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: pipeline_webscraping.pichau_tasks.Spark_Transformation_pichau manual__2024-10-29T15:14:07.538487+00:00 [queued]>
[2024-10-29T15:16:46.982+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: pipeline_webscraping.pichau_tasks.Spark_Transformation_pichau manual__2024-10-29T15:14:07.538487+00:00 [queued]>
[2024-10-29T15:16:46.983+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-10-29T15:16:46.999+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): pichau_tasks.Spark_Transformation_pichau> on 2024-10-29 15:14:07.538487+00:00
[2024-10-29T15:16:47.006+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=428) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-10-29T15:16:47.008+0000] {standard_task_runner.py:72} INFO - Started process 430 to run task
[2024-10-29T15:16:47.008+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'pipeline_webscraping', 'pichau_tasks.Spark_Transformation_pichau', 'manual__2024-10-29T15:14:07.538487+00:00', '--job-id', '73', '--raw', '--subdir', 'DAGS_FOLDER/DagWebScraping.py', '--cfg-path', '/tmp/tmpc0k4cczp']
[2024-10-29T15:16:47.011+0000] {standard_task_runner.py:105} INFO - Job 73: Subtask pichau_tasks.Spark_Transformation_pichau
[2024-10-29T15:16:47.056+0000] {task_command.py:467} INFO - Running <TaskInstance: pipeline_webscraping.pichau_tasks.Spark_Transformation_pichau manual__2024-10-29T15:14:07.538487+00:00 [running]> on host 4b5bd9addb3f
[2024-10-29T15:16:47.130+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='pipeline_webscraping' AIRFLOW_CTX_TASK_ID='pichau_tasks.Spark_Transformation_pichau' AIRFLOW_CTX_EXECUTION_DATE='2024-10-29T15:14:07.538487+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-29T15:14:07.538487+00:00'
[2024-10-29T15:16:47.131+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-10-29T15:16:47.151+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2024-10-29T15:16:47.151+0000] {spark_submit.py:327} ERROR - Failed to decode base64 keytab: Incorrect padding
[2024-10-29T15:16:47.154+0000] {spark_submit.py:304} INFO - Could not load connection string spark_default, defaulting to spark://spark:7077
[2024-10-29T15:16:47.156+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://spark:7077 --conf fs.s3a.access.key=AVcyU1dIcrfI88EqIa7p --conf fs.s3a.secret.key=****** --jars /opt/***/spark/jars/hadoop-aws-3.3.1.jar,/opt/***/spark/jars/aws-java-sdk-bundle-1.12.300.jar --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py --json_file_path s3a://rawzone/pichau/pichau_2024-10-27/data.json
[2024-10-29T15:16:49.201+0000] {spark_submit.py:634} INFO - Warning: Ignoring non-Spark config property: fs.s3a.access.key
[2024-10-29T15:16:49.201+0000] {spark_submit.py:634} INFO - Warning: Ignoring non-Spark config property: fs.s3a.secret.key
[2024-10-29T15:16:49.663+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:49 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-10-29T15:16:51.273+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SparkContext: Running Spark version 3.5.2
[2024-10-29T15:16:51.274+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SparkContext: OS info Linux, 5.15.153.1-microsoft-standard-WSL2, amd64
[2024-10-29T15:16:51.275+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SparkContext: Java version 17.0.13
[2024-10-29T15:16:51.301+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO ResourceUtils: ==============================================================
[2024-10-29T15:16:51.302+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-10-29T15:16:51.303+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO ResourceUtils: ==============================================================
[2024-10-29T15:16:51.303+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SparkContext: Submitted application: MyApp
[2024-10-29T15:16:51.325+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-10-29T15:16:51.333+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO ResourceProfile: Limiting resource is cpu
[2024-10-29T15:16:51.333+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-10-29T15:16:51.395+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SecurityManager: Changing view acls to: ***
[2024-10-29T15:16:51.396+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SecurityManager: Changing modify acls to: ***
[2024-10-29T15:16:51.396+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SecurityManager: Changing view acls groups to:
[2024-10-29T15:16:51.397+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SecurityManager: Changing modify acls groups to:
[2024-10-29T15:16:51.397+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-10-29T15:16:51.636+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO Utils: Successfully started service 'sparkDriver' on port 46627.
[2024-10-29T15:16:51.665+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SparkEnv: Registering MapOutputTracker
[2024-10-29T15:16:51.697+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SparkEnv: Registering BlockManagerMaster
[2024-10-29T15:16:51.716+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-10-29T15:16:51.717+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-10-29T15:16:51.720+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-10-29T15:16:51.751+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-488bbe90-4ff9-4ca7-986c-680a914e39a4
[2024-10-29T15:16:51.763+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-10-29T15:16:51.776+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-10-29T15:16:51.907+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-10-29T15:16:51.969+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:51 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2024-10-29T15:16:52.003+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO SparkContext: Added JAR file:///opt/***/spark/jars/hadoop-aws-3.3.1.jar at spark://4b5bd9addb3f:46627/jars/hadoop-aws-3.3.1.jar with timestamp 1730215011264
[2024-10-29T15:16:52.003+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO SparkContext: Added JAR file:///opt/***/spark/jars/aws-java-sdk-bundle-1.12.300.jar at spark://4b5bd9addb3f:46627/jars/aws-java-sdk-bundle-1.12.300.jar with timestamp 1730215011264
[2024-10-29T15:16:52.080+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark:7077...
[2024-10-29T15:16:52.122+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO TransportClientFactory: Successfully created connection to spark/172.18.0.3:7077 after 23 ms (0 ms spent in bootstraps)
[2024-10-29T15:16:52.459+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20241029151652-0000
[2024-10-29T15:16:52.469+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 40827.
[2024-10-29T15:16:52.469+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO NettyBlockTransferService: Server created on 4b5bd9addb3f:40827
[2024-10-29T15:16:52.471+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-10-29T15:16:52.479+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 4b5bd9addb3f, 40827, None)
[2024-10-29T15:16:52.483+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO BlockManagerMasterEndpoint: Registering block manager 4b5bd9addb3f:40827 with 434.4 MiB RAM, BlockManagerId(driver, 4b5bd9addb3f, 40827, None)
[2024-10-29T15:16:52.485+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 4b5bd9addb3f, 40827, None)
[2024-10-29T15:16:52.486+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 4b5bd9addb3f, 40827, None)
[2024-10-29T15:16:52.528+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241029151652-0000/0 on worker-20241029144356-172.18.0.6-37177 (172.18.0.6:37177) with 6 core(s)
[2024-10-29T15:16:52.530+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO StandaloneSchedulerBackend: Granted executor ID app-20241029151652-0000/0 on hostPort 172.18.0.6:37177 with 6 core(s), 1024.0 MiB RAM
[2024-10-29T15:16:52.788+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:52 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-10-29T15:16:53.091+0000] {spark_submit.py:634} INFO - <pyspark.sql.session.SparkSession object at 0x7f19ba7a40b0>
[2024-10-29T15:16:53.099+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:53 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-10-29T15:16:53.101+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:53 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2024-10-29T15:16:53.177+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:53 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241029151652-0000/0 is now RUNNING
[2024-10-29T15:16:55.008+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: s3a://rawzone/pichau/pichau_2024-10-28/data.json.
[2024-10-29T15:16:55.009+0000] {spark_submit.py:634} INFO - java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2024-10-29T15:16:55.010+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2688)
[2024-10-29T15:16:55.011+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:3431)
[2024-10-29T15:16:55.012+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3466)
[2024-10-29T15:16:55.012+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:174)
[2024-10-29T15:16:55.013+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3574)
[2024-10-29T15:16:55.013+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3521)
[2024-10-29T15:16:55.014+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:540)
[2024-10-29T15:16:55.015+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
[2024-10-29T15:16:55.015+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:53)
[2024-10-29T15:16:55.015+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:366)
[2024-10-29T15:16:55.016+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2024-10-29T15:16:55.017+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2024-10-29T15:16:55.017+0000] {spark_submit.py:634} INFO - at scala.Option.getOrElse(Option.scala:189)
[2024-10-29T15:16:55.018+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2024-10-29T15:16:55.018+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.json(DataFrameReader.scala:362)
[2024-10-29T15:16:55.019+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-10-29T15:16:55.019+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-10-29T15:16:55.020+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-10-29T15:16:55.021+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-10-29T15:16:55.021+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-10-29T15:16:55.022+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-10-29T15:16:55.022+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-10-29T15:16:55.023+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-10-29T15:16:55.023+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-10-29T15:16:55.024+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-10-29T15:16:55.024+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-10-29T15:16:55.025+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-10-29T15:16:55.026+0000] {spark_submit.py:634} INFO - Caused by: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2024-10-29T15:16:55.026+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2592)
[2024-10-29T15:16:55.028+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2686)
[2024-10-29T15:16:55.029+0000] {spark_submit.py:634} INFO - ... 26 more
[2024-10-29T15:16:55.072+0000] {spark_submit.py:634} INFO - Erro ao ler o arquivo JSON: An error occurred while calling o32.json.
[2024-10-29T15:16:55.073+0000] {spark_submit.py:634} INFO - : java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2024-10-29T15:16:55.073+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2688)
[2024-10-29T15:16:55.073+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:3431)
[2024-10-29T15:16:55.074+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3466)
[2024-10-29T15:16:55.074+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:174)
[2024-10-29T15:16:55.074+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3574)
[2024-10-29T15:16:55.075+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3521)
[2024-10-29T15:16:55.075+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:540)
[2024-10-29T15:16:55.075+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
[2024-10-29T15:16:55.076+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$1(DataSource.scala:724)
[2024-10-29T15:16:55.076+0000] {spark_submit.py:634} INFO - at scala.collection.immutable.List.map(List.scala:293)
[2024-10-29T15:16:55.077+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.checkAndGlobPathIfNecessary(DataSource.scala:722)
[2024-10-29T15:16:55.077+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.checkAndGlobPathIfNecessary(DataSource.scala:551)
[2024-10-29T15:16:55.078+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:404)
[2024-10-29T15:16:55.078+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2024-10-29T15:16:55.079+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2024-10-29T15:16:55.079+0000] {spark_submit.py:634} INFO - at scala.Option.getOrElse(Option.scala:189)
[2024-10-29T15:16:55.080+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2024-10-29T15:16:55.080+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.json(DataFrameReader.scala:362)
[2024-10-29T15:16:55.080+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-10-29T15:16:55.081+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-10-29T15:16:55.081+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-10-29T15:16:55.081+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-10-29T15:16:55.082+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-10-29T15:16:55.082+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-10-29T15:16:55.083+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-10-29T15:16:55.083+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-10-29T15:16:55.084+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-10-29T15:16:55.084+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-10-29T15:16:55.084+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-10-29T15:16:55.085+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-10-29T15:16:55.085+0000] {spark_submit.py:634} INFO - Caused by: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2024-10-29T15:16:55.086+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2592)
[2024-10-29T15:16:55.087+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2686)
[2024-10-29T15:16:55.087+0000] {spark_submit.py:634} INFO - ... 29 more
[2024-10-29T15:16:55.087+0000] {spark_submit.py:634} INFO - 
[2024-10-29T15:16:55.088+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-10-29T15:16:55.088+0000] {spark_submit.py:634} INFO - File "/opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py", line 218, in <module>
[2024-10-29T15:16:55.088+0000] {spark_submit.py:634} INFO - transformation_data(spark,args.json_file_path)
[2024-10-29T15:16:55.089+0000] {spark_submit.py:634} INFO - File "/opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py", line 77, in transformation_data
[2024-10-29T15:16:55.089+0000] {spark_submit.py:634} INFO - df=df.withColumn('value',df['value'].cast(DecimalType(12,2)))
[2024-10-29T15:16:55.089+0000] {spark_submit.py:634} INFO - ^^
[2024-10-29T15:16:55.090+0000] {spark_submit.py:634} INFO - UnboundLocalError: cannot access local variable 'df' where it is not associated with a value
[2024-10-29T15:16:55.147+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO SparkContext: Invoking stop() from shutdown hook
[2024-10-29T15:16:55.148+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-10-29T15:16:55.163+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO SparkUI: Stopped Spark web UI at http://4b5bd9addb3f:4040
[2024-10-29T15:16:55.182+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-10-29T15:16:55.182+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-10-29T15:16:55.213+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-10-29T15:16:55.240+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO MemoryStore: MemoryStore cleared
[2024-10-29T15:16:55.242+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO BlockManager: BlockManager stopped
[2024-10-29T15:16:55.256+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-10-29T15:16:55.263+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-10-29T15:16:55.296+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO SparkContext: Successfully stopped SparkContext
[2024-10-29T15:16:55.297+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO ShutdownHookManager: Shutdown hook called
[2024-10-29T15:16:55.297+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO ShutdownHookManager: Deleting directory /tmp/spark-4c4089fb-713b-4af1-89f5-3b052f643ba6
[2024-10-29T15:16:55.316+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO ShutdownHookManager: Deleting directory /tmp/spark-4c4089fb-713b-4af1-89f5-3b052f643ba6/pyspark-087c1127-60ff-4d03-996b-7b4148c0affc
[2024-10-29T15:16:55.329+0000] {spark_submit.py:634} INFO - 24/10/29 15:16:55 INFO ShutdownHookManager: Deleting directory /tmp/spark-1b73141a-0b29-45b6-a508-0517bd0995ac
[2024-10-29T15:16:55.399+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --conf fs.s3a.access.key=AVcyU1dIcrfI88EqIa7p --conf fs.s3a.secret.key=****** --jars /opt/***/spark/jars/hadoop-aws-3.3.1.jar,/opt/***/spark/jars/aws-java-sdk-bundle-1.12.300.jar --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py --json_file_path s3a://rawzone/pichau/pichau_2024-10-27/data.json. Error code is: 1.
[2024-10-29T15:16:55.404+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=pipeline_webscraping, task_id=pichau_tasks.Spark_Transformation_pichau, run_id=manual__2024-10-29T15:14:07.538487+00:00, execution_date=20241029T151407, start_date=20241029T151646, end_date=20241029T151655
[2024-10-29T15:16:55.418+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-10-29T15:16:55.418+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 73 for task pichau_tasks.Spark_Transformation_pichau (Cannot execute: spark-submit --master spark://spark:7077 --conf fs.s3a.access.key=AVcyU1dIcrfI88EqIa7p --conf fs.s3a.secret.key=****** --jars /opt/***/spark/jars/hadoop-aws-3.3.1.jar,/opt/***/spark/jars/aws-java-sdk-bundle-1.12.300.jar --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py --json_file_path s3a://rawzone/pichau/pichau_2024-10-27/data.json. Error code is: 1.; 430)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --conf fs.s3a.access.key=AVcyU1dIcrfI88EqIa7p --conf fs.s3a.secret.key=****** --jars /opt/***/spark/jars/hadoop-aws-3.3.1.jar,/opt/***/spark/jars/aws-java-sdk-bundle-1.12.300.jar --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py --json_file_path s3a://rawzone/pichau/pichau_2024-10-27/data.json. Error code is: 1.
[2024-10-29T15:16:55.452+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-10-29T15:16:55.474+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-29T15:16:55.480+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
